### PR TITLE
storage: Be smart about default filesystem type

### PR DESCRIPTION
**Pull request by mvollmer**

### CODE REVIEW

File: pkg/storaged/block/format-dialog.jsx [hunk 1]
```diff
@@ -196,6 +209,24 @@ function format_dialog_internal(client, path, start, size, enable_dos_extended,
...
+    let default_type = null;
+    if (block.IdUsage == "filesystem" && is_supported(block.IdType))
+        default_type = block.IdType;
+    else {
+        const root_block = find_root_fsys_block();
+        console.log("ROOT", root_block);
+        if (root_block && is_supported(root_block.IdType))
+            default_type = root_block.IdType;
+        else if (client.anaconda?.default_fsys_type && is_supported(client.anaconda.default_fsys_type))
+            default_type = client.anaconda.default_fsys_type;
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:225]: These 3 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240118-125446-dfad5288-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx [hunk 2]
```diff
@@ -154,6 +156,17 @@ export function format_dialog(client, path, start, size, enable_dos_extended) {
     }
 }
 
+function find_root_fsys_block() {
+    const root = client.anaconda?.mount_point_prefix || "/";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:160]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240118-161616-eaf4cdd4-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 1 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:225]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240118-161616-eaf4cdd4-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

File: pkg/storaged/block/format-dialog.jsx (hunk 2 above)
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:160]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240119-080732-cee85e18-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:160]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240119-134403-75b4b530-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
The code looks good to me, but I have questions about changing the default behavior from recommending xfs to now silently falling back to ext4. I wonder what the argument for the default being ext4 is? I know xfs is the RHEL blessed fs.

**Review by jelly — DISMISSED** [review]

**Comment by mvollmer** [comment]
> I wonder what the argument for the default being ext4 is? I know xfs is the RHEL blessed fs.

This fallback should be exceedingly rare. Normally, the default is taken from the root filesystem, which is xfs on RHEL.

Only if Cockpit does not support the filesystem type of the root filesystem (or can't find the root fs due to bugs) will it fall back to ext4.

We can of course have a different fallback since it shouldn't really matter. I chose ext4 because I think it's the most standard Linux filesystem.

File: pkg/storaged/block/format-dialog.jsx [hunk 3]
```diff
@@ -154,6 +156,17 @@ export function format_dialog(client, path, start, size, enable_dos_extended) {
...
+    for (const p in client.blocks) {
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:161]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240122-144509-75b4b530-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Comment by jelly** [comment]
Since my approval the pixel tests have changed, I'm happy to re-approve once that's fixed.

File: pkg/storaged/block/format-dialog.jsx [hunk 4]
```diff
@@ -155,6 +157,17 @@ export function format_dialog(client, path, start, size, enable_dos_extended) {
     }
 }
 
+function find_root_fsys_block() {
+    const root = client.anaconda?.mount_point_prefix || "/";
```
Comment (cockpituous) [inline pkg/storaged/block/format-dialog.jsx:161]: This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19854-20240123-141158-23f39417-fedora-39-devel//Coverage/lcov/github-pr.diff.gcov.html)

**Review by jelly — APPROVED** [review]

**Pull request merged**